### PR TITLE
Added code to not add MaxExecutionTimeHintToQuery if already present

### PR DESCRIPTION
**Pull request by AltamashShaikh**
### Description:

Added code to not add MaxExecutionTimeHintToQuery if already present
Fixes: #PG-3106

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: tests/PHPUnit/Unit/DbHelperTest.php [hunk 1]
```diff
@@ -74,6 +74,7 @@ public function getTestQueries()
             ['SELECT  /*+ MAX_EXECUTION_TIME(1500) */  * FROM table', 'SELECT * FROM table', 1.5],
             ['SELECT  /*+ MAX_EXECUTION_TIME(20000) */  column FROM (SELECT * FROM table)', 'SELECT column FROM (SELECT * FROM table)', 20],
             ['SELECT * FROM table', 'SELECT * FROM table', 0],
+            ['SELECT /*+ MAX_EXECUTION_TIME(1000) */ * FROM table', 'SELECT /*+ MAX_EXECUTION_TIME(1000) */ * FROM table', 3.5], //Should not append MAX_EXECUTION_TIME as 3500 if already present
```
Comment (michalkleiner) [inline tests/PHPUnit/Unit/DbHelperTest.php]: Where does the 3.5 come from?

Comment (AltamashShaikh) [reply tests/PHPUnit/Unit/DbHelperTest.php]: @michalkleiner Its there to test the behaviour to not append `/*+ MAX_EXECUTION_TIME(3500) */`, Any plugin or method sets a `MAX_EXECUTION_TIME` and we have 3.5 seconds set as `archiving_query_max_execution_time:3500` config setting but it should not append and rather use the value set by plugins or any other method

File: core/DbHelper.php [hunk 2]
```diff
@@ -300,7 +300,8 @@ public static function addMaxExecutionTimeHintToQuery($sql, $limit)
 
         $sql = trim($sql);
         $pos = stripos($sql, 'SELECT');
-        if ($pos !== false) {
+        $isMaxExecutionTimeoutAlreadyPresent = (stripos($sql, 'MAX_EXECUTION_TIME(') !== False);
```
Comment (bx80) [inline core/DbHelper.php]:
```suggestion
        $isMaxExecutionTimeoutAlreadyPresent = (stripos($sql, 'MAX_EXECUTION_TIME(') !== false);
```

**Review by bx80 — CHANGES_REQUESTED** [review]

File: tests/PHPUnit/Unit/DbHelperTest.php (hunk 1 above)
Comment (michalkleiner) [reply tests/PHPUnit/Unit/DbHelperTest.php]: Right, ok, thanks. Now I see the number actually doesn't matter in this particular test, can be anything.

Comment (michalkleiner) [reply tests/PHPUnit/Unit/DbHelperTest.php]:
```suggestion
            ['SELECT /*+ MAX_EXECUTION_TIME(1000) */ * FROM table', 'SELECT /*+ MAX_EXECUTION_TIME(1000) */ * FROM table', 3.5], // should not append/change MAX_EXECUTION_TIME hint if already present
```

**Review by michalkleiner — APPROVED** [review]

**Review by bx80 — APPROVED** [review]

**Comment by AltamashShaikh** [comment]
@bx80 / @michalkleiner Can you please merge this PR

**Pull request merged**